src/sortStudents.ts

```diff
@@ -1,16 +1,60 @@
 
 export interface Student {
   // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
+
 }
 
 export enum SortType {
   // describe SortType enum
+  Name = 'n',
+  Surname = 's',
+  Age = 'a',
+  Married = 'm',
+  AverageGrade = 'aG',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(students: Student[],
+  sortBy: SortType, order: SortOrder): Student[] {
+  const sortedStudents = [...students];
+
+  sortedStudents.sort((a, b) => {
+    let result = 0;
+
+    switch (sortBy) {
+      case SortType.Name:
+        result = a.name.localeCompare(b.name);
+        break;
+      case SortType.Surname:
+        result = a.surname.localeCompare(b.surname);
+        break;
+      case SortType.Age:
+        result = a.age - b.age;
+        break;
+
+      case SortType.Married:
+        result = Number(a.married) - Number(b.married);
+        break;
```

Comment:
booleans can be converted to numbers 1 and 0
so you can combine these two cases too
```suggestion
      case SortType.Age:
      case SortType.Married:
        result = Number(a[sortBy]) - Number(b[sortBy]);
        break;
```